File: Java/Working with Numbers/34. Decimal to Octal conversion.java
import java.util.Scanner;

class Main {
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.println("Enter the Decimal value:");
        int decimal = userInput.nextInt();
        
        // String octal = "";
        // if(decimal == 0) {
        //     octal = "0";
        // }
        // while(decimal > 0) {
        //     int rem = decimal % 8;
        //     octal = rem + octal;
        //     decimal /= 8;
        // }

        //Built-in Method 
        String octal = Integer.toOctalString(decimal);
        System.out.println("The Octal value of " + decimal + " is " + octal );

        userInput.close();
    }
}

File: Python/02. Working with Numbers/34. Decimal to Octal conversion.py
decimal = int(input("Enter the Decimal Value :"))
octal = oct(decimal)[2:] # slicing '0b' from output

print("Octal Number of",decimal, "is",octal)

# Method 2

# octal = ""
# if decimal == 0:
#     octal = "0"

# while decimal > 0:
#     rem = decimal % 8
#     octal = str(rem) + octal
#     decimal //= 8

# print(octal)